src/robot/can_adapter/adapter.py

```diff
@@ -0,0 +1,52 @@
+"""
+CAN adapter module.
+"""
+import asyncio
+from typing import Callable, Dict
+
+import can
+
+
+class CANAdapter:
+    """
+    This adapter abstracts away sending and receving messages on the CAN bus.
+    """
+
+    def __init__(self, can_bus: can.Bus):
+        self._can_bus = can_bus
+        self._notifier = can.Notifier(can_bus, [],
+                                      loop=asyncio.get_event_loop())
+
+    def stop(self) -> None:
+        """
+        Stop routing messages to the handlers. 
+        """
+        self._notifier.stop()
+
+    def send(self, arbitration_id: int, data: bytes) -> None:
+        """
+        Send a message on the bus.
+        """
+        self._can_bus.send(can.Message(arbitration_id=arbitration_id,
```

Comment:
is_extended_id should either be set to False, or be set by a setter/constructor set member variable.

Reply:
Should probably also set check=True, to check if generated message is a valid CAN message. It will raise Errors if wrong parameters are given.

Reply:
Agreed!